# Exercice : Analyse des données de résistance du béton avec des dictionnaires

### Lecture des données

Le cellule suivante lit les données de résistance du béton et les stocke dans une liste de listes. Chaque sous-liste contient les données d'une ligne du fichier.


In [8]:
file_path = "./resources/Concrete_Data_Yeh/Concrete_Data_Yeh.csv"

# lecture du fichier

dict_tableau = {}
with open(file_path, 'r') as file:
    for line_id, line_content in enumerate(file):
        if line_id == 0:
            header = line_content.strip().split(',')
        else:
            data = line_content.strip().split(',')
            data_dict = dict(zip(header, data))
            dict_tableau[data_dict['cement']] = data_dict
            del data_dict['cement']
dict_tableau
            

    



{'540': {'slag': '0',
  'flyash': '0',
  'water': '173',
  'superplasticizer': '0',
  'coarseaggregate': '1125',
  'fineaggregate': '613',
  'age': '3',
  'csMPa': '41.64'},
 '332.5': {'slag': '142.5',
  'flyash': '0',
  'water': '228',
  'superplasticizer': '0',
  'coarseaggregate': '932',
  'fineaggregate': '594',
  'age': '7',
  'csMPa': '30.28'},
 '198.6': {'slag': '132.4',
  'flyash': '0',
  'water': '192',
  'superplasticizer': '0',
  'coarseaggregate': '978.4',
  'fineaggregate': '825.5',
  'age': '3',
  'csMPa': '9.13'},
 '266': {'slag': '112',
  'flyash': '87',
  'water': '178',
  'superplasticizer': '10',
  'coarseaggregate': '910',
  'fineaggregate': '745',
  'age': '28',
  'csMPa': '39.42'},
 '380': {'slag': '95',
  'flyash': '0',
  'water': '228',
  'superplasticizer': '0',
  'coarseaggregate': '932',
  'fineaggregate': '594',
  'age': '7',
  'csMPa': '32.82'},
 '475': {'slag': '0',
  'flyash': '0',
  'water': '162',
  'superplasticizer': '9.5',
  'coarseaggregate': '1044'

### Afficher uniquement que les clés du dictionnaire.

In [ ]:
## todo

### Afficher uniquement les valeurs du dictionnaire.

In [9]:
# todo

### Récupérer une valeur d'un dictionnaire 

Affiche l'age du ciment dont le mélange est 332.5 kg dans 1 m3.

In [ ]:
# todo

### Supprimer d'une entrée 

Supprimer du dictionnaire le mélange  est 198.6 kg dans 1 m3. Afficher les clés du dictonnaire après la suppression.


In [ ]:
# todo

Supprimer du dctionnaire l'age du ciment dont le mélange est 310.0 kg dans 1 m3. Afficher les valeurs pour ce mélange après la suppression.

In [11]:
# todo

### Mettre à jour une valeur dans un dictionnaire

Modifier la valeur de l'age du ciment (multiplier par 2) dont le mélange est 389.9 kg dans 1 m3. Afficher la valeur de l'age du ciment après la modification.

In [ ]:
# todo

### Ajouter une entrée dans un dictionnaire

Ajouter une entrée dans le dictionnaire pour le mélange  est 999 kg dans 1 m3 avec les données suiavntes : 

| Cement | Slag | Fly Ash | Water | Superplasticizer | Coarse Aggregate | Fine Aggregate | Age | csMPa |
|--------|------|---------|-------|------------------|------------------|----------------|-----|-------|
| 999    | 0    | 0       | 162   | 2.5              | 1040             | 676            | 28  | 79.99 |


Afficher les cette nouvelle partie dictionnaire après l'ajout.


In [ ]:
# todo